pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -669,14 +678,22 @@ private BrokerResponseNative handleRequest(long requestId, String query,
       //       can always list the running queries and cancel query again until it ends. Just that such race
       //       condition makes cancel API less reliable. This should be rare as it assumes sending queries out to
       //       servers takes time, but will address later if needed.
-      QueryServers queryServers = _queriesById.computeIfAbsent(requestId, k -> new QueryServers(query));
+      _queriesById.put(requestId, new QueryServers(query, offlineRoutingTable, realtimeRoutingTable));
       LOGGER.debug("Keep track of running query: {}", requestId);
-      queryServers.addServers(offlineRoutingTable, realtimeRoutingTable);
```

Comment:
one reason to use addServers() was to accumulate all servers involved when running a query, which might do some subqueries (thus recursively calling this handleRequest() method (where I assumed different routing tables might be created for some subqueries...)

Reply:
Yes, and that is the reason why I also moved the `_queriesById.remove(requestId)` into the `handleRequest()` to avoid this scenario

Reply:
Make sense!